### PR TITLE
#11 Add mailbox set

### PR DESCRIPTION
**Pull request by oubchid**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -153,6 +161,60 @@ export interface IMailboxGetResponse {
...
+  | 'tooManyChanges'
+  | 'unknownCapability'
+  | 'unknownMethod'
+  | 'unsupportedFilter'
+  | 'unsupportedSort'
+  | 'willDestroy';
+
+export interface ISetError {
+  type: IErrorType;
+  description?: string;
```
Comment (chibenwa) [inline src/types.ts:215]: invalidProperties may have a properties field as well.

**Review by chibenwa — COMMENTED** [review]
No tests?

**Review by alagane — DISMISSED** [review]

**Review by fabienmoyon — APPROVED** [review]

File: src/types.ts [hunk 2]
```diff
@@ -15,11 +15,11 @@ export interface IGetArguments<Properties> {
...
 
-export interface ISetArguments<CreatedObject> {
+export interface ISetArguments<Foo> {
   accountId: string;
-  ifInState?: string | null;
-  create?: { [temporaryId: string]: CreatedObject };
-  update?: { [id: string]: { [jsonPointer: string]: any } };
+  ifInState?: string;
+  create?: { [id: string]: Partial<Foo> };
+  update?: Partial<Foo>;
```
Comment (alagane) [inline src/types.ts:22]:
```suggestion
  ifInState?: string | null;
  create?: { [temporaryId: string]: Partial<Foo> };
  update?: Partial<Foo> & { [id: string]: { [jsonPointer: string]: any } };
```

Comment (alagane) [reply src/types.ts]: I did I mistake, the Partial<Foo> should be just before the `{ [jsonPointer: string]: any }`

**Review by alagane — APPROVED** [review]

**Review by fabienmoyon — APPROVED** [review]

**Pull request merged**